daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/SequenceGroup.scala

```diff
@@ -215,20 +216,37 @@ abstract class SequenceGroupTermBase(xml: Node, lexicalParent: SchemaComponent,
     }
   }
 
-  private lazy val checkMembersHaveUniqueNamesInNamespaces: Unit = {
-    val childrenGroupedByQName = groupMembers.groupBy { gm =>
-      // previous checks should ensure that all group members are either local
-      // elements or element references
-      Assert.invariant(gm.isInstanceOf[ElementBase])
-      gm.asInstanceOf[ElementBase].namedQName
-    }
-    childrenGroupedByQName.foreach { case (qname, children) =>
-      if (children.length > 1) {
+  private lazy val checkUnorderedSequenceMembersHaveUniqueNamesInNamespaces: Unit = {
+    // previous checks should ensure that all group members for unordered sequences are either local
+    // elements or element references
+    groupChildrenBy(_.namedQName)
```

Comment:
I forget why we added the `groupChildrenBy` function, but it only replaces `elementChildren.groupBy`. The former isn't significantly shorter and I don't think it is any more clear. Is it worth removing `groupChildrenBy`?

Reply:
It used to do alot more than elementChildren.groupBy, but I moved out the filter stuff, since that wasn't related to grouping